### ISSUE TITLE
[Monitoring V2] Add Linux node-selectors / tolerations

### ISSUE THREAD

**Issue by aiyengar2** [rancher/rancher#31498]
Adds nodeSelectors and tolerations throughout all components deployed by Monitoring V2 to ensure that they are never deployed on a Windows host.

After this issue is merged, we should ensure that no monitoring component is deployed on a Windows host within a Windows cluster.

**Comment by sowmyav27** [comment]
To validate:
- Deploy windows cluster oon 2.5-head
- deploy Monitoring v2
- Verify the monitoring related workloads are deployed _only_ on Linux nodes and not on windows nodes
- Verify Monitoring v2 works for linux nodes _only_

**Comment by thehejik** [comment]
Monitoring helm chart was in version `9.4.204-rc04` and not marked in the dropdown as `Linux only` so does it mean that for windows v2 monitoring support user is supposed to install also extra chart `rancher-windows-exporter`? Also please what is the purpose `rancher-wins-upgrader`, is it somehow related to v2 monitoring?

**Comment by aiyengar2** [comment]
@thehejik 

> It seems there is nothing related to v2 monitoring running on windows nodes, please @aiyengar2 is it correct?

For testing this specific issue, we should select the Cluster Type on the UI manually to be RKE (since Cluster Type RKE Windows won't exist until https://github.com/rancher/dashboard/issues/2537 is resolved) and deploy. But you are correct that the expectation for this issue is that nothing related to v2 monitoring should be running on Windows nodes, so if that is the case after RKE exporters are deployed we can close out this specific issue.

w.r.t. [`rancher-windows-exporter`](https://github.com/rancher/rancher/issues/31497) and [`rancher-wins-upgrader`](https://github.com/rancher/rancher/issues/31504), those are being prioritized in separate issues linked here.

TLDR for their purpose:
- rancher-windows-exporter is the new Helm subchart that will support deploying https://github.com/prometheus-community/windows_exporter (the equivalent of node_exporter for Windows nodes) via a Docker image located in https://github.com/rancher/windows_exporter-package. This is identical in concept to Monitoring V1 exporter-windows, which similarly deployed wmi_exporter (the older name for windows_exporter) based on the Docker image located in https://github.com/rancher/wmi_exporter-package, except it utilizes a newer version of wins that enables metrics to be scraped from the Pod's IP instead of the node's IP
- rancher-wins-upgrader is a Helm-based solution that allows Windows clusters to upgrade the wins version and configuration across nodes in a Windows cluster. The primary use case is to allow a 1 click upgrade of Windows clusters provisioned before 2.5.8 to the wins version necessary to run Windows Monitoring V2 (instead of having to log into each host, cleanup the host, and then re-add it to the cluster OR log into each host, download the new wins binary / config, and restart wins), but it can also be used in general to make it easier to whitelist new process hostpaths or proxy additional ports that are now supported by `wins cli proxy`

**Comment by thehejik** [comment]
After selecting `cluster.provider.rke` from the dropdown the v2 monitoring-rancher chart is also successfully deployed (with a bit different set of pods than before with `cluster.provider.other`) and nothing is running on windows nodes:

```
k get all -n cattle-monitoring-system -o custom-columns=KIND:.kind,NAME:.metadata.name,NODE:.spec.nodeName,PHASE:.status.phase,IMAGES:.spec.containers[*].image --sort-by=.kind
KIND          NAME                                                      NODE               PHASE     IMAGES
DaemonSet     pushprox-kube-scheduler-client                            <none>             <none>    <none>
DaemonSet     pushprox-kube-etcd-client                                 <none>             <none>    <none>
DaemonSet     rancher-monitoring-prometheus-node-exporter               <none>             <none>    <none>
DaemonSet     pushprox-kube-controller-manager-client                   <none>             <none>    <none>
DaemonSet     pushprox-kube-proxy-client                                <none>             <none>    <none>
Deployment    rancher-monitoring-operator                               <none>             <none>    <none>
Deployment    rancher-monitoring-prometheus-adapter                     <none>             <none>    <none>
Deployment    rancher-monitoring-grafana                                <none>             <none>    <none>
Deployment    pushprox-kube-proxy-proxy                                 <none>             <none>    <none>
Deployment    pushprox-kube-scheduler-proxy                             <none>             <none>    <none>
Deployment    pushprox-kube-etcd-proxy                                  <none>             <none>    <none>
Deployment    pushprox-kube-controller-manager-proxy                    <none>             <none>    <none>
Deployment    rancher-monitoring-kube-state-metrics                     <none>             <none>    <none>
Pod           rancher-monitoring-prometheus-node-exporter-9g7m2         ip-172-31-7-131    Running   rancher/mirrored-prom-node-exporter:v1.0.1
Pod           rancher-monitoring-operator-6657c9cbd7-xvcw4              ip-172-31-7-131    Running   rancher/mirrored-coreos-prometheus-operator:v0.38.1,rancher/mirrored-squareup-ghostunnel:v1.5.2
Pod           rancher-monitoring-prometheus-adapter-69bbff57cc-nbkc4    ip-172-31-7-131    Running   rancher/mirrored-directxman12-k8s-prometheus-adapter-amd64:v0.7.0
Pod           rancher-monitoring-prometheus-node-exporter-2f5ct         ip-172-31-6-208    Running   rancher/mirrored-prom-node-exporter:v1.0.1
Pod           rancher-monitoring-kube-state-metrics-5574fffff-48vsq     ip-172-31-7-131    Running   rancher/mirrored-coreos-kube-state-metrics:v1.9.7
Pod           rancher-monitoring-prometheus-node-exporter-ws2dr         ip-172-31-15-163   Running   rancher/mirrored-prom-node-exporter:v1.0.1
Pod           pushprox-kube-etcd-client-fpzvm                           ip-172-31-15-163   Running   rancher/pushprox-client:v0.1.0-rancher1-client
Pod           prometheus-rancher-monitoring-prometheus-0                ip-172-31-7-131    Running   rancher/mirrored-prom-prometheus:v2.18.2,rancher/mirrored-coreos-prometheus-config-reloader:v0.38.1,rancher/mirrored-jimmidyson-configmap-reload:v0.3.0,rancher/mirrored-library-nginx:1.19.2-alpine
Pod           pushprox-kube-controller-manager-client-slk9r             ip-172-31-6-208    Running   rancher/pushprox-client:v0.1.0-rancher1-client
Pod           rancher-monitoring-grafana-7fbfcdff6f-rqcqr               ip-172-31-7-131    Running   rancher/mirrored-kiwigrid-k8s-sidecar:0.1.151,rancher/mirrored-grafana-grafana:7.1.5,rancher/mirrored-library-nginx:1.19.2-alpine
Pod           pushprox-kube-scheduler-proxy-778649899-zr6ww             ip-172-31-7-131    Running   rancher/pushprox-proxy:v0.1.0-rancher1-proxy
Pod           pushprox-kube-scheduler-client-5v6z4                      ip-172-31-6-208    Running   rancher/pushprox-client:v0.1.0-rancher1-client
Pod           pushprox-kube-proxy-proxy-645f6c6d9-dfzgf                 ip-172-31-7-131    Running   rancher/pushprox-proxy:v0.1.0-rancher1-proxy
Pod           pushprox-kube-proxy-client-th7dq                          ip-172-31-6-208    Running   rancher/pushprox-client:v0.1.0-rancher1-client
Pod           pushprox-kube-proxy-client-fwvwj                          ip-172-31-7-131    Running   rancher/pushprox-client:v0.1.0-rancher1-client
Pod           pushprox-kube-proxy-client-frkcs                          ip-172-31-15-163   Running   rancher/pushprox-client:v0.1.0-rancher1-client
Pod           alertmanager-rancher-monitoring-alertmanager-0            ip-172-31-7-131    Running   rancher/mirrored-prom-alertmanager:v0.21.0,rancher/mirrored-jimmidyson-configmap-reload:v0.3.0
Pod           pushprox-kube-etcd-proxy-544b98d789-t5p4j                 ip-172-31-7-131    Running   rancher/pushprox-proxy:v0.1.0-rancher1-proxy
Pod           pushprox-kube-controller-manager-proxy-6d7546f885-v7pp6   ip-172-31-7-131    Running   rancher/pushprox-proxy:v0.1.0-rancher1-proxy
ReplicaSet    pushprox-kube-controller-manager-proxy-6d7546f885         <none>             <none>    <none>
ReplicaSet    pushprox-kube-etcd-proxy-544b98d789                       <none>             <none>    <none>
ReplicaSet    pushprox-kube-scheduler-proxy-778649899                   <none>             <none>    <none>
ReplicaSet    rancher-monitoring-grafana-7fbfcdff6f                     <none>             <none>    <none>
ReplicaSet    rancher-monitoring-kube-state-metrics-5574fffff           <none>             <none>    <none>
ReplicaSet    rancher-monitoring-operator-6657c9cbd7                    <none>             <none>    <none>
ReplicaSet    rancher-monitoring-prometheus-adapter-69bbff57cc          <none>             <none>    <none>
ReplicaSet    pushprox-kube-proxy-proxy-645f6c6d9                       <none>             <none>    <none>
Service       rancher-monitoring-alertmanager                           <none>             <none>    <none>
Service       rancher-monitoring-prometheus-node-exporter               <none>             <none>    <none>
Service       rancher-monitoring-prometheus-adapter                     <none>             <none>    <none>
Service       rancher-monitoring-prometheus                             <none>             <none>    <none>
Service       alertmanager-operated                                     <none>             <none>    <none>
Service       rancher-monitoring-operator                               <none>             <none>    <none>
Service       rancher-monitoring-kube-state-metrics                     <none>             <none>    <none>
Service       rancher-monitoring-grafana                                <none>             <none>    <none>
Service       pushprox-kube-scheduler-proxy                             <none>             <none>    <none>
Service       pushprox-kube-scheduler-client                            <none>             <none>    <none>
Service       pushprox-kube-proxy-proxy                                 <none>             <none>    <none>
Service       pushprox-kube-proxy-client                                <none>             <none>    <none>
Service       pushprox-kube-etcd-proxy                                  <none>             <none>    <none>
Service       pushprox-kube-etcd-client                                 <none>             <none>    <none>
Service       pushprox-kube-controller-manager-client                   <none>             <none>    <none>
Service       prometheus-operated                                       <none>             <none>    <none>
Service       pushprox-kube-controller-manager-proxy                    <none>             <none>    <none>
StatefulSet   prometheus-rancher-monitoring-prometheus                  <none>             <none>    <none>
StatefulSet   alertmanager-rancher-monitoring-alertmanager              <none>             <none>    <none>
```

Once `windowsExporter.enabled=true` is set a DaemonSet `rancher-monitoring-windows-exporter` and corresponding pods and services are deployed in addition. These pods are running on windows nodes.
